common/src/main/java/com/redhat/lightblue/rest/RestConfiguration.java

```diff
@@ -74,7 +74,7 @@ public static LightblueFactory getFactory() {
         return factory;
     }
 
-    public static LightblueFactory getFactory(
+    public synchronized static LightblueFactory getFactory(
```

Comment:
You can use synchronize on RestConfiguration.class.
On Feb 10, 2016 2:16 PM, "Dennis Crissman" <notifications@github.com> wrote:

> In common/src/main/java/com/redhat/lightblue/rest/RestConfiguration.java
> <https://github.com/lightblue-platform/lightblue-rest/pull/205#discussion_r52526173>
> :
>
> > @@ -74,7 +74,7 @@ public static LightblueFactory getFactory() {
> >          return factory;
> >      }
> >
> > -    public static LightblueFactory getFactory(
> > +    public synchronized static LightblueFactory getFactory(
>
> synchronized(this)
>
> That was my initial thought also, but we are in a static context so it
> won't work. I could create a lock object also in a volatile state, but I
> guess that seemed silly over just making the method synchronized.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/lightblue-platform/lightblue-rest/pull/205/files#r52526173>
> .
>